### DISCUSSION TITLE
Add instant bypass to sites which are eligble

### DISCUSSION THREAD

**Discussion by ccuser44** [FastForwardTeam/FastForward#830] [Ideas]
(*Re-posted this from #828*)

The current behavior of FastForward bypasses is that they run after the HTML dom has been loaded.

Now for some websites this is unnecessary and redundant (for example because the destination URL can be derived from the middle URL). In fact this is what the bit.ly bypass does, however unfortunately this cannot be used for any other bypasses.

How this should function that before the browser even makes a connection to the website it's already redirected to the destination.

This would make the user experience much better. Oftentimes there is annoying load times and the intermediate page appears for a few seconds.
It would save unnecessary bandwidth usage, not to mention improve privacy and prevent potential malicious javascript from running (adf.ly wink wink).

(Now of course this should only be enabled on sites which it can work on and be a feature for custom bypasses)

**Comment by ghost** [comment]
Do you have a proof of concept for a site (adf.ly is a good one to try)?
